packages/runtime/src/templates/server.ts

```diff
@@ -0,0 +1,73 @@
+import https from 'https'
+
+import { NodeRequestHandler, Options } from 'next/dist/server/next-server'
+
+import { getNextServer, NextServerType } from './handlerUtils'
+
+const NextServer: NextServerType = getNextServer()
+
+interface NetlifyNextServerOptions extends Options {
+  netlifyRevalidateToken: string
+}
+
+class NetlifyNextServer extends NextServer {
+  private netlifyRevalidateToken?: string
+
+  public constructor(options: NetlifyNextServerOptions) {
+    super(options)
+    this.netlifyRevalidateToken = options.netlifyRevalidateToken
+  }
+
+  public getRequestHandler(): NodeRequestHandler {
+    const handler = super.getRequestHandler()
+    return async (req, res, parsedUrl) => {
+      if (req.headers['x-prerender-revalidate']) {
+        if (this.netlifyRevalidateToken) {
+          await this.netlifyRevalidate(req.url)
```

Comment:
Will this revalidate both the `html` page and the `json` page, e.g. `/posts/1` and `/_next/data/xtcOaw2OikW05zh4TC3_7/posts/1.json`?

Reply:
It won't at the moment, you're right. The next piece of work, for the next milestone, is to generate the right paths for data routes, i18n and RSC.